docs/operations/versioning.md

```diff
@@ -0,0 +1,305 @@
+
+
+`Version Maintenance` helps with the handling of `maintained` and `unmaintained` versions, specifically:
+ - Enforces one `supported` version per `maintained` **minor version** (versus only one `supported` version without `Version Maintenance`)
+   - be aware that if there is no supported version for a maintained minor version, it will pick the **latest** version independent of the classification and sets it to `supported`.
+ - Deprecates all versions with an `unmaintained` minor (no matter what classification they have).
+ - Sets the correct deprecation date for `unmaintained` versions (`controllers.cloudProfile.kubernetesVersionManagement.expirationDuration`).
+ - Does not set an expiration date on the latest version with an `unmaintained` version. Otherwise, an undesired and automated forceful minor Kubernetes version upgrade is possible. This is potentially harmful for the deployed workload.  
+
+Because Shoot clusters using an `unmaintained` minor version potentially have a higher security risk, Gardener will update these clusters after a certain date / grace-period.
+By defining the field `expirationDurationUnmaintainedVersion` in the Gardener Controller Manager configuration ([see below](#gardener-controller-manager-configuration-for-version-maintenance)), Gardener can set a different expiration date for 
+`unmaintained` versions than for `maintained` versions (the expiration date for `maintained` versions can be configured by the field `controllers.cloudProfile.kubernetesVersionManagement.expirationDuration`). 
+For example `maintained` versions are deprecated with an expiration of four months, while `unmaintained` already expire in one. 
+As described above, expired versions will be forcefully updated by Gardener.
+
+### Gardener Controller Manager configuration for Version Maintenance
+
+The Gardener Controller Manager has to be configured to enable the optional `Version Maintenance`.
+This is an example Gardener Controller Manager configuration enabling `Version Maintenance` for Kubernetes, but not for Machine image versions:
+
+``` yaml
+apiVersion: controllermanager.config.gardener.cloud/v1alpha1
+kind: ControllerManagerConfiguration
+controllers:
+  cloudProfile:
+    concurrentSyncs: 5
+    kubernetesVersionManagement:
+      expirationDuration: 2880h # defaults to 120 days / 4 months
+      versionMaintenance:
+        maintainedKubernetesVersions: 3
+        expirationDurationUnmaintainedVersion: 720h
+    machineImageVersionManagement:
+      expirationDuration: 2880h # defaults to 120 days / 4 months
+```
+
+## Duties of the Gardener Operator
+
+Even though Gardener strives for high automation, we think that it leads to safer operations 
+having a human operator in charge of the following operations: 
+
+- Adding `preview` versions
+- Promotion of a `preview` version to `supported`. We think this should be done on a case-by-case basis, hence Gardener deliberately does not offer an automation in this case.
```

Comment:
Hm, I don't like this too much. Yes, it's safer this way of course, but given the frequency of patch releases this might involve quite a lot of manual action...

Reply:
Thats one of the cases we discussed a while back with @vlerenc - we do not want to have an automation in place that could be harmful in case we "forget"about it. Lets say there is a preview version that has not been checked thoroughly - nevertheless after 2 weeks it will be promoted to supported ...